File: rrouter-core-old/src/main/java/org/sheedon/rrouter/DefaultRspConverter.java
package org.sheedon.rrouter;

import org.sheedon.rrouter.core.support.IRspModel;

/**
 * 默认响应结果转换器
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/2/15 7:41 下午
 */
public class DefaultRspConverter extends Converter.Factory {

    @Override
    public Converter<?, IRspModel<?>> createCheckConverter() {
        return DefaultConverter.INSTANCE;
    }

    static class DefaultConverter implements Converter<Object, IRspModel<?>> {

        private static final DefaultConverter INSTANCE = new DefaultConverter();

        @Override
        public IRspModel<?> convert(Object o) {
            if (o == null) {
                return null;
            }
            return (IRspModel<?>) o;
        }


    }
}

File: rrouter-core/src/main/java/org/sheedon/rrouter/core/DefaultRspConverter.kt
package org.sheedon.rrouter.core

/**
 * 默认响应结果转换器
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/2/15 7:41 下午
 */
class DefaultRspConverter : Converter.Factory() {

    override fun createCheckConverter(): Converter<*, IRspModel<*>?> {
        return DefaultConverter.INSTANCE
    }

    internal class DefaultConverter : Converter<Any?, IRspModel<*>?> {
        override fun convert(t: Any?): IRspModel<*>? {
            return if (t == null) {
                null
            } else t as IRspModel<*>
        }

        companion object {
            val INSTANCE = DefaultConverter()
        }
    }

}